R/apa_print_merMod.R

```diff
@@ -1,144 +1,112 @@
-#'   } else {
-#'     res_table <- res_table[, est_cols]
-#'   }
-#'   # p values for glmms seem to be possible, but are calculated from z values
-#'
-#'   variable_labels(res_table) <- c(
-#'     term = "Term"
-#'     , estimate = "$b$"
-#'     , std.err = "$\\mathit{SE}$"
-#'     , conf.int = paste0(args_confint$level * 100, "% CI")
-#'     , statistic = "$t$"
-#'     , df = "$\\mathit{df}$"
-#'     , p.value = "$p$"
-#'   )[colnames(res_table)]
-#'   class(res_table) <- c("apa_results_table", "data.frame")
-#'
-#'   # prepare final output container ----
-#'   res <- init_apa_results()
-#'   res$table <- res_table
-#'
-#'   res$estimate <- as.list(
-#'     paste0("$b = ", res_table$estimate, "$, ", unlist(print_cis))
-#'   )
-#'   # if(in_paren) res$estimate <- in_paren(res$estimate) # not necessary, yet
-#'
-#'   if(isLmerTest) {
-#'     res$statistic <- as.list(paste0("$t(", res_table$df, ") = ", res_table$statistic, "$, $p ", add_equals(res_table$p.value), "$"))
-#'   } else {
-#'     res$statistic <- as.list(paste0("$t = ", res_table$statistic, "$"))
-#'   }
-#'   if(in_paren) res$statistic <- in_paren(res$statistic)
-#'
-#'   res$full_result <- as.list(paste0(res$estimate, ", ", res$statistic))
-#'
-#'   names(res$full_result) <- names(res$statistic) <- names(res$estimate) <- sanitize_terms(res_table$term)
-#'
-#'   # return
-#'   res
-#' }
-#'
-#' #' @rdname apa_print.merMod
-#' #' @method apa_print mixed
-#' #' @export
-#'
-#' apa_print.mixed <- function(x, ...) {
-#'
-#'   anova_table <- x$anova_table
-#'   attr(anova_table, "method") <- attr(x, "method")
-#'   apa_print(anova_table, ...)
-#' }
+#' Format Statistics from Hierarchical (Generalized) Linear Models (APA 6th edition)
+#'
+#' These methods take objects from various R functions that calculate
+#' hierarchical (generalized) linear models to create formatted character strings
+#' to report the results in accordance with APA manuscript guidelines.
+#'
+#' @param x A fitted hierarchical (generalized) linear model, either from [lme4::lmer()],
+#'   [lmerTest::lmer()], [afex::mixed()], or [lme4::glmer()].
+#' @param args_confint Named list. Additional arguments that are passed to [lme4::confint.merMod()], see details.
+#' @param est_name     An optional character. The label to be used for fixed-effects coefficients.
+#' @inheritParams print_anova
+#' @param ... Further arguments that are passed to [printnum()].
+#' @details
+#'   Confidence intervals are calculated by calling [lme4::confint.merMod()]. By
+#'   default, *profile* confidence intervals are calculated, except for generalized
+#'   linear models with non-fixed scale. In such cases, *bootstrap* confidence intervals
+#'   are calculated by default. See [lme4::confint.merMod()] for details.
+#'
+#' @family apa_print
+#' @rdname apa_print.merMod
+#' @method apa_print merMod
+#' @export
+#' @md
+
+apa_print.merMod <- function(
+  x
+  , args_confint
+  , in_paren = FALSE
+  , est_name
```

Comment:
In most other methods we define `est_name = NULL` as default. Any particular reason to deviate from this here? I like the `NULL`-default because it makes explicit, that no value needs to be passed.

Reply:
Same for `args_confint`.

Reply:
No explicit reason, will change it to `NULL`

Reply:
done